From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Praschl <praschlpaul@g-p.at>
Date: Sun, 2 Aug 2026 23:01:48 +0200
Subject: [PATCH 092/117] arm64: dts: apple: Add iPhone 7 Bluetooth

D111 wires its BCM4355C0 controller to UART1 with hardware flow
control, PMIC shutdown and device wake GPIOs. Use the BCM4345C5
fallback to retain the tested generic lifecycle.

Signed-off-by: Paul Praschl <praschlpaul@g-p.at>

Orchard-replay-of: dae61536e19f99aa581de49c944a4cbdd17bdca4
---
 arch/arm64/boot/dts/apple/t8010-7.dtsi | 30 +++++++++++++++++++++++++-
 1 file changed, 29 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/apple/t8010-7.dtsi b/arch/arm64/boot/dts/apple/t8010-7.dtsi
index c01225bc2..50372e7e6 100644
--- a/arch/arm64/boot/dts/apple/t8010-7.dtsi
+++ b/arch/arm64/boot/dts/apple/t8010-7.dtsi
@@ -12,6 +12,10 @@
 / {
 	chassis-type = "handset";
 
+	aliases {
+		bluetooth0 = &bluetooth;
+	};
+
 	gpio-keys {
 		compatible = "gpio-keys";
 
@@ -72,7 +76,7 @@
 			compatible = "apple,d2333-pmic-gpio";
 			gpio-controller;
 			#gpio-cells = <2>;
-			gpio-reserved-ranges = <0 10>, <11 10>;
+			gpio-reserved-ranges = <0 8>, <9 1>, <12 9>;
 		};
 
 		rtc@500 {
@@ -98,3 +102,27 @@
 		};
 	};
 };
+
+&pinctrl_ap {
+	bluetooth_uart1_pins: bluetooth-uart1-pins {
+		pinmux = <APPLE_PINMUX(96, 1)>,
+			 <APPLE_PINMUX(97, 1)>,
+			 <APPLE_PINMUX(98, 1)>,
+			 <APPLE_PINMUX(99, 1)>;
+	};
+};
+
+&serial1 {
+	pinctrl-0 = <&bluetooth_uart1_pins>;
+	pinctrl-names = "default";
+	uart-has-rtscts;
+	status = "okay";
+
+	bluetooth: bluetooth {
+		compatible = "brcm,bcm4355c0-bt", "brcm,bcm4345c5";
+		current-speed = <115200>;
+		max-speed = <1500000>;
+		shutdown-gpios = <&pmu_gpio 8 GPIO_ACTIVE_HIGH>;
+		device-wakeup-gpios = <&pinctrl_ap 87 GPIO_ACTIVE_HIGH>;
+	};
+};
